### PR TITLE
Added new alerts for kubernetes-storage

### LINKED ISSUE #10965: Prometheus Alerts for Storage are outdated
I recently compared our current alerts with the alerts managed by the community of https://github.com/kubernetes-monitoring/kubernetes-mixin and it seams they changed the used metrics and alerts quite a bit. They also added a alerts if PVCs are stuck in some pending state what could be helpful for ops. 

Current rules:
https://github.com/kubermatic/kubermatic/blob/87074a55c404f220777abd9e302887316d5aa2f5/charts/monitoring/prometheus/rules/general-kube-kubelet.yaml#L29
```yaml
      - alert: KubePersistentVolumeUsageCritical
        annotations:
          message: The PersistentVolume claimed by {{ $labels.persistentvolumeclaim }} in namespace {{ $labels.namespace }} is only {{ printf "%0.0f" $value }}% free.
          runbook_url: https://docs.kubermatic.com/kubermatic/master/cheat-sheets/alerting-runbook/#alert-kubepersistentvolumeusagecritical
        expr: |
          100 * kubelet_volume_stats_available_bytes{job="kubelet"}
            /
          kubelet_volume_stats_capacity_bytes{job="kubelet"}
            < 3
        for: 1m
        labels:
          severity: critical
          service: kubelet
          resource: "{{ $labels.namespace }}/{{ $labels.persistentvolumeclaim }}"
      - alert: KubePersistentVolumeFullInFourDays
        annotations:
          message: Based on recent sampling, the PersistentVolume claimed by {{ $labels.persistentvolumeclaim }} in namespace {{ $labels.namespace }} is expected to fill up within four days. Currently {{ $value }} bytes are available.
          runbook_url: https://docs.kubermatic.com/kubermatic/master/cheat-sheets/alerting-runbook/#alert-kubepersistentvolumefullinfourdays
        expr: |
          (
            kubelet_volume_stats_used_bytes{job="kubelet"}
              /
            kubelet_volume_stats_capacity_bytes{job="kubelet"}
          ) > 0.85
          and
          predict_linear(kubelet_volume_stats_available_bytes{job="kubelet"}[6h], 4 * 24 * 3600) < 0
        for: 5m
        labels:
          severity: critical
          service: kubelet
          resource: "{{ $labels.namespace }}/{{ $labels.persistentvolumeclaim }}"
```

Jsonnet ruleset of kubernetes-mixin:
https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/master/alerts/storage_alerts.libsonnet

After I looked up at , the following rendered alerting rules, seams to work (adjusted from 3% left to 5%)

```yaml
- "name": "kubernetes-storage"
  "rules":
    - "alert": "KubePersistentVolumeFillingUp"
      "annotations":
        "description": "The PersistentVolume claimed by {{ $labels.persistentvolumeclaim }} in Namespace {{ $labels.namespace }} is only {{ $value | humanizePercentage }} free."
        "runbook_url": "https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubepersistentvolumefillingup"
        "summary": "PersistentVolume is filling up."
      "expr": |
        (
          kubelet_volume_stats_available_bytes{job="kubelet"}
            /
          kubelet_volume_stats_capacity_bytes{job="kubelet"}
        ) < 0.05
        and
        kubelet_volume_stats_used_bytes{job="kubelet"} > 0
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_access_mode{ access_mode="ReadOnlyMany"} == 1
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_labels{label_excluded_from_alerts="true"} == 1
      "for": "1m"
      "labels":
        "severity": "critical"
    - "alert": "KubePersistentVolumeFillingUp"
      "annotations":
        "description": "Based on recent sampling, the PersistentVolume claimed by {{ $labels.persistentvolumeclaim }} in Namespace {{ $labels.namespace }} is expected to fill up within four days. Currently {{ $value | humanizePercentage }} is available."
        "runbook_url": "https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubepersistentvolumefillingup"
        "summary": "PersistentVolume is filling up."
      "expr": |
        (
          kubelet_volume_stats_available_bytes{job="kubelet"}
            /
          kubelet_volume_stats_capacity_bytes{job="kubelet"}
        ) < 0.15
        and
        kubelet_volume_stats_used_bytes{job="kubelet"} > 0
        and
        predict_linear(kubelet_volume_stats_available_bytes{job="kubelet"}[6h], 4 * 24 * 3600) < 0
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_access_mode{ access_mode="ReadOnlyMany"} == 1
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_labels{label_excluded_from_alerts="true"} == 1
      "for": "1h"
      "labels":
        "severity": "warning"
    - "alert": "KubePersistentVolumeInodesFillingUp"
      "annotations":
        "description": "The PersistentVolume claimed by {{ $labels.persistentvolumeclaim }} in Namespace {{ $labels.namespace }} only has {{ $value | humanizePercentage }} free inodes."
        "runbook_url": "https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubepersistentvolumeinodesfillingup"
        "summary": "PersistentVolumeInodes are filling up."
      "expr": |
        (
          kubelet_volume_stats_inodes_free{job="kubelet"}
            /
          kubelet_volume_stats_inodes{job="kubelet"}
        ) < 0.03
        and
        kubelet_volume_stats_inodes_used{job="kubelet"} > 0
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_access_mode{ access_mode="ReadOnlyMany"} == 1
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_labels{label_excluded_from_alerts="true"} == 1
      "for": "1m"
      "labels":
        "severity": "critical"
    - "alert": "KubePersistentVolumeInodesFillingUp"
      "annotations":
        "description": "Based on recent sampling, the PersistentVolume claimed by {{ $labels.persistentvolumeclaim }} in Namespace {{ $labels.namespace }} is expected to run out of inodes within four days. Currently {{ $value | humanizePercentage }} of its inodes are free."
        "runbook_url": "https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubepersistentvolumeinodesfillingup"
        "summary": "PersistentVolumeInodes are filling up."
      "expr": |
        (
          kubelet_volume_stats_inodes_free{job="kubelet"}
            /
          kubelet_volume_stats_inodes{job="kubelet"}
        ) < 0.15
        and
        kubelet_volume_stats_inodes_used{job="kubelet"} > 0
        and
        predict_linear(kubelet_volume_stats_inodes_free{job="kubelet"}[6h], 4 * 24 * 3600) < 0
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_access_mode{ access_mode="ReadOnlyMany"} == 1
        unless on(namespace, persistentvolumeclaim)
        kube_persistentvolumeclaim_labels{label_excluded_from_alerts="true"} == 1
      "for": "1h"
      "labels":
        "severity": "warning"
    - "alert": "KubePersistentVolumeErrors"
      "annotations":
        "description": "The persistent volume {{ $labels.persistentvolume }} has status {{ $labels.phase }}."
        "runbook_url": "https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubepersistentvolumeerrors"
        "summary": "PersistentVolume is having issues with provisioning."
      "expr": |
        kube_persistentvolume_status_phase{phase=~"Failed|Pending",job="kube-state-metrics"} > 0
      "for": "5m"
      "labels":
        "severity": "critical"
```

Potential we should also add 2 steps of alerts:
1. Warning at 7%
2. Critical at  5%

### PR DESCRIPTION
**Pull request by vishnueg**
Signed-off-by: vishnuegknr <vishnuegknr@gmail>

**What this PR does / why we need it**:
Add new alerts for kubernetes-storage
**Which issue(s) this PR fixes**:

Fixes #10965
We missed some of the alerts of PVC getting filled due to wrong alert expressions

**What type of PR is this?**
This will add new alerts in Prometheus
/kind cleanup

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by vishnueg** [comment]
@wurbanski  , I see there is a difference in the field of new alerts, "Message" has been replaced with "description" . Kindly check that especially and let me know the feedback if any correction needed.

**Comment by wurbanski** [comment]
>  "Message" has been replaced with "description"

Please use the name of the field that's used throughout KKP as we have automations based on the field name.

**Comment by vishnueg** [comment]
> > "Message" has been replaced with "description"
> 
> Please use the name of the field that's used throughout KKP as we have automations based on the field name.

>
> Correction has been done as per the terms used in kubermatic repo

File: charts/monitoring/prometheus/rules/src/general/kube-kubelet.yaml [hunk 1]
```diff
@@ -25,44 +25,121 @@ groups:
...
+        annotations:
+          message: The persistent volume {{ $labels.persistentvolume }} has status {{ $labels.phase }}.
+          runbook_url: https://docs.kubermatic.com/kubermatic/master/cheat-sheets/alerting-runbook/#alert-name-kubepersistentvolumeerrors
+          summary: PersistentVolume is having issues with provisioning.
+        expr: |
+          kube_persistentvolume_status_phase{phase=~"Failed|Pending",job="kube-state-metrics"} > 0
         for: 5m
         labels:
           severity: critical
-          service: kubelet
```
Comment (wurbanski) [inline charts/monitoring/prometheus/rules/src/general/kube-kubelet.yaml:64]: Please add these labels as well, they are used in alerta for service matching.

**Review by wurbanski — CHANGES_REQUESTED** [review]

File: charts/monitoring/prometheus/rules/src/general/kube-kubelet.yaml (hunk 1 above)
Comment (vishnueg) [reply charts/monitoring/prometheus/rules/src/general/kube-kubelet.yaml:64]: Labels added.

**Review by wurbanski — APPROVED** [review]

**Pull request merged**